source/js/engine.js

```diff
@@ -1,4 +1,18 @@
-function calculateProfitsOrLossesForEachStock(stocks) {
+function Engine(currencySign, commission) {
+  this.currencySign = currencySign;
+  this.commission = commission;
+}
```

Comment:
Take a look at **Delegation Theory**: https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/ch6.md#class-theory